### PR TITLE
feat(edit-json): detect indentation

### PR DESCRIPTION
**Pull request by ivorisoutdoors**
I looked at npm to see how they handled indenting and found a couple packages that detect indentation. I also included the JSON parser they use which gives better error messages on invalid json.

### CODE REVIEW

File: src/Handlers/EditJsonActionHandler.ts [hunk 1]
```diff
@@ -45,19 +46,23 @@ export class EditJsonActionHandler implements ActionHandlerContract<'edit-json'>
 
       try {
         Logger.info(`Reading ${file}.`);
-        let content = fs.readJsonSync(targetFile);
+        let content = fs.readFileSync(targetFile).toString();
+        if (content.charCodeAt(0) === 0xFEFF) {
+          content = content.slice(1)
+        }
```
Comment (innocenzi) [inline src/Handlers/EditJsonActionHandler.ts]: Can you explain what is it for? 

Apparently `0xfeff` is a zero-width space, but why would it need to be removed? Would it cause issue with `json-parse-better-errors`?

**Review by innocenzi — COMMENTED** [review]
Looks okay to me, even though I'd rather not add `json-parse-better-errors` to the bundle. Do you really think it's necessary? It's not that heavy, but I'm wondering about its usefulness.

File: src/Handlers/EditJsonActionHandler.ts (hunk 1 above)
Comment (ivorisoutdoors) [reply src/Handlers/EditJsonActionHandler.ts]: I copied some of this from npm. It also looks like jsonfile (which `fs-extra` uses) does something similar

**Comment by ivorisoutdoors** [comment]
@innocenzi `json-parse-better-errors` probably isn't very useful in this context. I can remove and use JSON.parse instead

**Comment by innocenzi** [comment]
Yeah, I'd rather only add `detect-indent`. Once you remove `json-parse-better-errors`, I'll be happy to merge. 👍

**Comment by ivorisoutdoors** [comment]
@innocenzi removed

**Comment by innocenzi** [comment]
Thank you! 

[v0.1.26](https://github.com/use-preset/use-preset/releases/tag/v0.1.26)

**Pull request merged**